src/utils/message.ts

```diff
@@ -1,22 +1,111 @@
-import { BlockfrostClientError, BlockfrostServerError, Responses } from '@blockfrost/blockfrost-js';
+import { BlockfrostClientError, BlockfrostServerError } from '@blockfrost/blockfrost-js';
 import { serializeError } from 'serialize-error';
-import { Messages } from '../types/message.js';
-import * as TxTypes from '../types/transactions.js';
-import { UtxosWithBlockResponse } from '../types/address.js';
-import { AccountInfo, BalanceHistoryData, ServerInfo } from '../types/response.js';
+import { Details, MessageId, Messages } from '../types/message.js';
+import { Responses } from '../types/response.js';
 import { logger } from './logger.js';
+import { Ajv } from 'ajv';
 
-export const getMessage = (message: string): Messages | undefined => {
-  try {
-    const parsedMessage: Messages = JSON.parse(message);
+export class MessageError extends Error {
+  constructor(public reason: string) {
```

Comment:
```suggestion
  constructor(reason: string) {
```

Can't we omit the `reason` prop from the MessageError? It seems it will always just duplicate the default `message` field eg:

```
{
  "id": 10,
  "type": "error",
  "data": {
    "error": {
      "reason": "[{\"instancePath\":\"\",\"schemaPath\":\"#/required\",\"keyword\":\"required\",\"params\":{\"missingProperty\":\"details\"},\"message\":\"must have required property 'details'\"}]",
      "name": "Error",
      "message": "[{\"instancePath\":\"\",\"schemaPath\":\"#/required\",\"keyword\":\"required\",\"params\":{\"missingProperty\":\"details\"},\"message\":\"must have required property 'details'\"}]"
    }
  }
}
```

Reply:
If I remember correctly, the value of the error.message attribute is `{error.constructor.name}: ${message}`; I'd like to avoid the error proposed to the user to not start with a meaningless 'MessageError: '.

Reply:
Yes, with latest refactoring it is no longer needed

Reply:
12963fc should solve this